### PR TITLE
[Merged by Bors] - avoid relocking head during builder health check

### PR DESCRIPTION
**Pull request by int88**
## Issue Addressed

#4314 

## Proposed Changes

avoid relocking head during builder health check

## Additional Info

NA

### CODE REVIEW

**Comment by int88** [comment]
@michaelsproul PTAL.

File: beacon_node/beacon_chain/src/beacon_chain.rs [hunk 1]
```diff
@@ -5707,13 +5707,9 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
     /// Since we are likely calling this during the slot we are going to propose in, don't take into
     /// account the current slot when accounting for skips.
     pub fn is_healthy(&self, parent_root: &Hash256) -> Result<ChainHealth, Error> {
+        let cache_head = self.canonical_head.cached_head();
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/src/beacon_chain.rs]:
```suggestion
        let cached_head = self.canonical_head.cached_head();
```

Comment (int88) [reply beacon_node/beacon_chain/src/beacon_chain.rs]: @michaelsproul updated!

**Review by michaelsproul — APPROVED** [review]
Even though I wasn't able to confirm a performance-benefit from this through the (limited) benchmarking I did, I think it's a safe change to make and a nice cleanup

**Review by paulhauner — APPROVED** [review]
I like this too! I doubt it has performance benefits (like Michael said) but it's definitely less prone to weird race conditions.

**Pull request closed without merging**